src/main/java/mate/academy/dao/MovieDaoImpl.java

```diff
@@ -0,0 +1,53 @@
+package mate.academy.dao;
+
+import java.util.Optional;
+import mate.academy.exceptions.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.Movie;
+import mate.academy.util.HibernateUtil;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+
+@Dao
+public class MovieDaoImpl implements MovieDao {
+    @Override
+    public Movie add(Movie movie) {
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.save(movie);
+            transaction.commit();
+            session.close();
+            return movie;
```

Comment:
```suggestion
            transaction.commit();
            return movie;
```
you're gonna close the session in the finally block